File: day01/solve.R
#!/usr/bin/env Rscript
d <- read.table("input.txt")[, 1]

for (i in d) {
  complement <- 2020 - i
  if (complement %in% d && i != complement) {
    print(paste("Part 1 answer:", i * complement))
    break
  }
}

answered <- FALSE
for (i in d) {
  for (j in d) {
    complement <- 2020 - (i + j)
    if (complement %in% d && i * 3 != 2020) {
      print(paste("Part 2 answer:", i * j * complement))
      answered <- TRUE
      break
    }
  }
  if (answered) {
    break
  }
}

File: day01/solve.py
#!/usr/bin/env python3
from itertools import combinations
from functools import reduce
from operator import mul

with open('input.txt', 'r') as ifile:
    nums = {int(n) for n in ifile}

print("Part 1's answer:")
for a, b in combinations(nums, 2):
    if a + b == 2020:
        print(a * b)
        break

print("Part 2's answer:")
for combo in combinations(nums, 3):
    if sum(combo) == 2020:
        print(reduce(mul, combo))
        break
